docs/myriad-fine-grained-scaling.md

```diff
@@ -0,0 +1,43 @@
+# Myriad Fine Grained Scaling
+
+The objective of "Fine Grained Scaling" is to bring elasticity of resources between Myriad/YARN and other Mesos frameworks. The idea is to allow YARN to take a resource offer from mesos, run enough containers (YARN tasks) that the offer can hold and release the offer once the containers finish.
+
+### The feature (currently) works as follows
+
+* Admin launches Node Managers via [Myriad flexup API](API.md), but with **zero** capacities.
+* The Node Managers report **zero** capacities to the Resource Manager upon registration.
+* A user submits an application to YARN (for e.g. a MapReduce job).
+* The application is added to RM's scheduling pipeline. However, YARN scheduler (for e.g. FairShareScheduler) cannot schedule the application containers as all the Node Managers have zero available capacity.
+* The Myriad Mesos Framework (running inside RM) subsequently receives resource offers from Mesos.
+* An offer (from a slave node that's already running a NM) is projected to YARN scheduler as "available capacity" on the Node Manager.
+* The YARN scheduler now goes ahead and allocates containers (tasks) for the Node Managers.
+* For each allocated container, Myriad framework spins up a "placeholder" mesos task. (In reality, a bunch of "placeholder" tasks are launched in a single shot as YARN scheduler allocates a bunch of containers as a result of increase in NM's capacity due to a single mesos offer.)
+* NMs become aware of container allocations via YARN's HB mechanism. But Myriad ensures that NMs are made aware of container allocations only after the corresponding "placeholder" mesos tasks are launched.
+* When NMs report to RM that some of the containers have "finished", Myriad sends out "finished" status updates to Mesos for the corresponding "placeholder" tasks.
+* Mesos takes back the resources from Myriad that were previously blocked using "placeholder" tasks.
+
+### To try out..
+
+* [Build myriad](myriad-dev.md) with a special "zero" profile added to myriad-config-default.yml:
+```
+profiles:
+  zero:
+    cpu: 0
+    mem: 0
+```
+* Spin up Resource Manager with "Myriad Scheduler" plugged into it.
```

Comment:
This is my first time through the code.  I didn't see any way we're differentiating these from other profiles.  Would a medium profile or large still be eligible for resizing?  That may be a feature as then there's some immediate capacity for containers.